### DISCUSSION TITLE
Improve custom JsonConverters by allowing to call the default Read/Write implementation in System.Text.Json

### DISCUSSION THREAD

**Discussion by LazaroOnline** [dotnet/runtime#46434] [Ideas]
When trying to implement a custom JsonConverter in `System.Text.Json`, sometimes you only want to implement one of the 2 methods `Read` or `Write`, or you may want to call the default implementation (as if the converter that you are writing was not there) before further processing the data.

But calling their default implementation is not currently possible as one would expect without issues or overly complicated solutions as seen in [this StackOverflow question](https://stackoverflow.com/questions/65430420/how-to-use-default-serialization-in-a-custom-system-text-json-jsonconverter ).

If you try to call the "default" like this:
```c#
public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
{
  return JsonSerializer.Deserialize<T>(ref reader, options);
}
public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
{
  return JsonSerializer.Serialize(value);
}
```
Then you may fall in an infinite loop if you try to add the converter as an attribute to a class definition, example:
```c#
[JsonConverter(typeof(CustomConverter<Person>))]
public class Person
{
	public string Name { get; set; }
	public int Age { get; set; }
}

public class CustomConverter<T> : JsonConverter<T>
{
    public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
      var value = JsonSerializer.Deserialize<T>(ref reader, options);
	  // Custom logic here...
	  return value;
    }
    public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
    {
      var json = JsonSerializer.Serialize(value);
	  // Custom logic here...
	  return json;
    }
}

[Fact]
public void CanDeserializeUsingClassAttribute()
{
	var name = "SomeName";
	var json = $@"{{ ""Name"": ""{name}"" }}";
	var result = JsonSerializer.Deserialize<Person>(json);
	// This should return the default JsonConverter implementation, but instead here is an infinite loop!
	//...
}
```

Can we have a way, provided by .Net runtime in System.Text.Json, to call the default implementation?
Something like:
```c#

public class CustomConverter<T> : JsonConverter<T>
{
    public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
    {
      return JsonDefaultSerialize<T>(ref reader, options);
    }
    public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
    {
      return JsonDefaultSerialize(value, options);
    }
}
```
So not only we can call leave the implementation to the default behaviour (as if the converter being written was not there), but also we could use the default implementation dynamically based on some condition.
Thank you!

**Comment by rdlaitila** [comment]
This solution worked for my use case related to this issue. My JsonConverter Write method looks as follows:

```csharp
public override void Write(Utf8JsonWriter writer, MyClass value, JsonSerializerOptions options)
{
	var doc = JsonSerializer.SerializeToDocument(value, value.GetType(), options);
	doc.WriteTo(writer);
}
```
